src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1497,6 +1497,16 @@ private void CheckRecordMemberNames(DiagnosticBag diagnostics)
             {
                 diagnostics.Add(ErrorCode.ERR_CloneDisallowedInRecord, member.Locations[0]);
             }
+
+            var declaredGetHashCodes = GetMembers("GetHashCode");
+            var isGetHashCodeNotDeclared = declaredGetHashCodes.IsEmpty;
+            var declaredEquals = GetMembers("Equals");
+            var isEqualsNotDeclared = declaredEquals.IsEmpty;
```

Comment:
I think this implementation is problematic. There *could* be user-defined members with these names which doesn't override Equals and GetHashCode methods.